ui/cryptomaterial/wallet_clickable_list.go

```diff
@@ -0,0 +1,116 @@
+package cryptomaterial
+
+import (
+	"gioui.org/layout"
+	"gioui.org/unit"
+	"github.com/crypto-power/cryptopower/libwallet/utils"
+)
+
+type ClickedItem struct {
+	AssetType utils.AssetType
+	Index     int
+}
+
+type WalletClickableList struct {
+	layout.List
+	theme           *Theme
+	clickables      []*Clickable
+	Radius          CornerRadius // this radius is used by the clickable
+	selectedItem    int
+	DividerHeight   unit.Dp
+	IsShadowEnabled bool
+	IsHoverable     bool
+	ItemIDs         map[*Clickable]ClickedItem
+}
+
+func (t *Theme) NewWalletClickableList(axis layout.Axis) *WalletClickableList {
+	click := &WalletClickableList{
+		theme:        t,
+		selectedItem: -1,
+		List: layout.List{
+			Axis: axis,
+		},
+		IsHoverable: true,
+	}
+
+	return click
+}
+
+func (cl *WalletClickableList) ItemClicked() (bool, ClickedItem) {
+	for _, clickable := range cl.clickables {
+		if clickable.Clicked() {
+			if itemID, exists := cl.ItemIDs[clickable]; exists {
+				return true, itemID
+			}
+		}
+	}
+	return false, ClickedItem{}
+}
+
+func (cl *WalletClickableList) handleClickables(itemIDs []ClickedItem) {
+	if len(cl.clickables) != len(itemIDs) {
+
+		// Resize the clickables slice to match the size of the provided itemIDs
+		cl.clickables = make([]*Clickable, len(itemIDs))
+
+		// Initialize the ItemIDs map
+		cl.ItemIDs = make(map[*Clickable]ClickedItem)
+
+		for i, itemID := range itemIDs {
+			clickable := cl.theme.NewClickable(cl.IsHoverable)
+			cl.clickables[i] = clickable
+			cl.ItemIDs[clickable] = itemID
+		}
+	}
+
+	// Handle clicked items (this was already part of your original handleClickables method)
+	for index, clickable := range cl.clickables {
+		for clickable.Clicked() {
+			cl.selectedItem = index
+		}
+	}
+}
+
+func (cl *WalletClickableList) Layout(gtx layout.Context, itemIDs []ClickedItem, w layout.ListElement) layout.Dimensions {
```

Comment:
This is same as what we have in `Clickable list`

```
func (cl *ClickableList) Layout(gtx layout.Context, count int, w layout.ListElement) layout.Dimensions {
	cl.handleClickables(count)
	return cl.List.Layout(gtx, count, func(gtx C, i int) D {
		if cl.IsShadowEnabled && cl.clickables[i].button.Hovered() {
			shadow := cl.theme.Shadow()
			shadow.SetShadowRadius(14)
			shadow.SetShadowElevation(5)
			return shadow.Layout(gtx, func(gtx C) D {
				return cl.row(gtx, count, i, w)
			})
		}
		return cl.row(gtx, count, i, w)
	})
}
```

Reply:
similar code but very different, especially when you try to merge the two, it introduces necessary complexity to other lists across the app , what do you suggest?